File: sieve-of-eratosthes.py
#This is just the converted Rust code to Python
import math
import sys

def is_prime(number):
  for i in range(2, int(math.sqrt(number))+1):
    if number % i == 0:
      return False
  return True

def main():

  print("Enter the limit: ")
  limit = int(input())

  numbers = list(range(2, limit+1)) 
  primes = []

  for number in numbers:
    if is_prime(number):
      primes.append(number)
      print(number)

  print(primes)

if __name__ == '__main__':
  main()

File: sieve_of_eratosthes.rs
//Implimenting a sieve of erastothene to find all prime numbers less than a given number.
//It should allow for user input to set the limit.


fn main (){
   let mut limit = String::new();
    println!("Enter the limit: ");
    std::io::stdin().read_line(&mut limit).unwrap();
    //using u32 for only positive integers
    let limit: u32 = limit.trim().parse().unwrap();
    //List of all numbers from 2 to the limit
    let mut numbers: Vec<u32> = (2..limit).collect();
    //List of all prime numbers
    let mut primes: Vec<u32> = Vec::new();

    //Iterate through the numbers list
    for number in numbers.iter(){
        //If the number is prime, add it to the primes list
        if is_prime(number){
            primes.push(*number);
        }
    } 
}
//Function to check if a number is prime
